### ISSUE TITLE
Wan 14B train problem！

### ISSUE THREAD

**Issue by mingtiannihao** [aigc-apps/VideoX-Fun#148]
I'm using DeepSpeed ZeRO Stage 3 + bf16 + xFormers + enable_text_encoder_in_dataloader + low_vram, with batch size = 1.
My input video is 81 frames of size 480×832, and I'm doing full fine-tuning.

Even with 8× A800 80GB GPUs, I'm still encountering OOM (Out of Memory) errors.

What could be the reason for this? Any suggestions on how to reduce memory usage further?

**Comment by hkunzhe** [comment]
@mingtiannihao Sorry for the late reply. Full parameter training of the 14B model requires enabling ZeRO-3 (cross-node sharding) in a multi-machine setup. We use at least 4 nodes (8 A800 80G).

**Comment by bubbliiiing** [comment]
You can try to train less module by setting trainable_modules in train.py to save GPU memory.

**Comment by mingtiannihao** [comment]
if args.train_mode != "normal":
                    mask = get_random_mask(new_examples["pixel_values"][-1].size(), image_start_only=False)
                    mask_pixel_values = new_examples["pixel_values"][-1] * (1 - mask) 
                    # Wan 2.1 use 0 for masked pixels
                    # + torch.ones_like(new_examples["pixel_values"][-1]) * -1 * mask
                    new_examples["mask_pixel_values"].append(mask_pixel_values)
                    new_examples["mask"].append(mask)
                    
                    clip_pixel_values = new_examples["pixel_values"][-1][0].permute(1, 2, 0).contiguous()
                    clip_pixel_values = (clip_pixel_values * 0.5 + 0.5) * 255
                    new_examples["clip_pixel_values"].append(clip_pixel_values)

**Comment by mingtiannihao** [comment]
In Wan 2.1, masked pixels are set to 0, but there's also a commented option to use -1.
Which is better for training?

**Comment by cindyn-nflx** [comment]
@mingtiannihao I am also having this problem. Did you find a fix?

**Comment by samuruph** [comment]
Do you have the settings to train with multiple nodes ?
